pwndbg/gdblib/disasm/x86.py

```diff
@@ -330,12 +330,18 @@ def _parse_memory(self, instruction: PwndbgInstruction, op: EnhancedOperand, emu
         # Get memory address (Ex: lea    rax, [rip + 0xd55], this would return $rip+0xd55. Does not dereference)
         target = 0
 
-        # There doesn't appear to be a good way to read from segmented
-        # addresses within GDB.
         if op.mem.segment != 0:
-            return None
-
-        if op.mem.base != 0:
+            if op.mem.segment == X86_REG_FS:
+                if (target := pwndbg.gdblib.regs.fsbase) is None:
+                    return None
+            elif op.mem.segment == X86_REG_GS:
+                if (target := pwndbg.gdblib.regs.gsbase) is None:
+                    return None
+            else:
+                return None
+        # Both a segment and base cannot be in use
+        # A Capstone bug sometimes sets base to RIP if segment is set
```

Comment:
What's with this bug? :p

Is it documented anywhere or sth? can we link?

What do we do in this case?

Reply:
I'll look to make a Capstone report of it.

This shows the capstone disassembly (using `cstool`) of `mov r9, qword ptr fs:[0x10]`. For some reason, the rip register is shown as base. I just happened to randomly catch this while stepping through a binary - I didn't see any issues on the Capstone GitHub, but I can make one.

![image](https://github.com/user-attachments/assets/0c0450e5-f3a6-421e-801b-ad9fc581969a)

Reply:
Oh nevermind - a mistake on my part. Totally glossed over the `rip` in the memory operand... This comment can be removed....